OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/alias/AliasSourceElement.scala

```diff
@@ -0,0 +1,280 @@
+
+    override def isMethodBound: Boolean = true
+
+    override def declaredMethod: DeclaredMethod = fieldReference.context.method
+
+    override def method: Method = fieldReference.context.method.definedMethod
+
+    override def isReferenceType: Boolean = fieldReference.field.fieldType.isReferenceType
+
+    override def referenceType: ReferenceType = fieldReference.field.fieldType.asReferenceType
+
+    override def isAliasField: Boolean = true
+
+    override def asAliasField: AliasField = this
+}
+
+/**
+ * Represents a static field that is part of an alias relation.
+ */
+case class AliasStaticField(field: Field) extends AliasSourceElement {
+
+    override def element: Field = field
+
+    override def isMethodBound: Boolean = false
+
+    override def isReferenceType: Boolean = field.fieldType.isReferenceType
+
+    override def referenceType: ReferenceType = field.fieldType.asReferenceType
+
+    override def isAliasStaticField: Boolean = true
+
+    override def asAliasStaticField: AliasStaticField = this
+
+}
+
+/**
+ * Represents a method return value of a method that is part of an alias relation.
+ */
+case class AliasReturnValue(override val method: Method, project: SomeProject) extends AliasSourceElement {
+
+    private[this] val dm = project.get(DeclaredMethodsKey)(method)
+
+    override def element: AnyRef = method
+
+    override def declaredMethod: DeclaredMethod = dm
+
+    override def isMethodBound: Boolean = true
+
+    override def isReferenceType: Boolean = method.returnType.isReferenceType
+
+    override def referenceType: ReferenceType = method.returnType.asReferenceType
+
+    override def isAliasReturnValue: Boolean = true
+
+    override def asAliasReturnValue: AliasReturnValue = this
+}
+
+/**
+ * Represents a formal parameter of a method that is part of an alias relation.
+ */
+case class AliasFormalParameter(fp: VirtualFormalParameter) extends AliasSourceElement {
+
+    override def element: VirtualFormalParameter = fp
+
+    override def method: Method = fp.method.definedMethod
+
+    override def declaredMethod: DeclaredMethod = fp.method
+
+    override def isMethodBound: Boolean = true
+
+    override def isReferenceType: Boolean = {
+        if (fp.parameterIndex == -1) fp.method.declaringClassType.isObjectType
+        else fp.method.definedMethod.parameterTypes(fp.parameterIndex).isReferenceType
+    }
+
+    override def referenceType: ReferenceType = {
+        if (fp.parameterIndex == -1) fp.method.declaringClassType.asReferenceType
+        else fp.method.definedMethod.parameterTypes(fp.parameterIndex).asReferenceType
+    }
+
+    override def isAliasFP: Boolean = true
+
+    override def asAliasFP: AliasFormalParameter = this
+}
+
+/**
+ * A persistent representation (using pcs instead of TAC value origins) for a UVar.
+ *
+ * @see [[org.opalj.tac.fpcf.analyses.cg.persistentUVar]]
+ */
+case class PersistentUVar(valueInformation: ValueInformation, defSites: IntTrieSet)
```

Comment:
This should not be part of this file, but be in a generic place. You might want to take the code from here:
https://github.com/opalj/opal/pull/1/files#diff-62df04527a62216c2330e018931c309e5a1b88dac0e66e6a11e451a89dcf0517
and the following files

Reply:
How should i handle that the other PR is not merged yet? Just copy&paste the relevant code and add it to this PR?

Reply:
Yes, that's probably the only practical solution for now. Unless that part of the code is in isolated commits (I didn't check that), then you could cherry-pick them.

Reply:
Cherry-picking this commit (https://github.com/opalj/opal/commit/c3b9715575a51b44bd97184b8acd8fd69c29af7c) should be sufficient. However, the class has been created in the tac package, therefore i can't use it as it is. I could move to the br package but i don't know if this is intended and should also be done in the other PR to avoid duplicates after merging both.

Reply:
I have cherry picked the commit now and moved the class to the br package with some neccessary adjustments. As mentioned, this should probably also be done in the other PR (if you want to keep it this way)